src/site/content/en/blog/async-functions/index.md

```diff
@@ -0,0 +1,392 @@
+    });
+}
+```
+
+And here's the same thing using async functions:
+
+```js
+async function logFetch(url) {
+  try {
+    const response = await fetch(url);
+    console.log(await response.text());
+  }
+  catch (err) {
+    console.log('fetch failed', err);
+  }
+}
+```
+
+It's the same number of lines, but all the callbacks are gone. This makes it way
+easier to read, especially for those less familiar with promises.
+
+{% Aside %}
+Anything you `await` is passed through `Promise.resolve()`, so you can
+safely `await` non-native promises.
+{% endAside %}
+
+## Async return values
+
+Async functions *always* return a promise, whether you use `await` or not. That
+promise resolves with whatever the async function returns, or rejects with
+whatever the async function throws. So with:
+
+```js
+// wait ms milliseconds
+function wait(ms) {
+  return new Promise(r => setTimeout(r, ms));
+}
+
+async function hello() {
+  await wait(500);
+  return 'world';
+}
+```
+
+…calling `hello()` returns a promise that *fulfills* with `"world"`.
+
+```js
+async function foo() {
+  await wait(500);
+  throw Error('bar');
+}
+```
+
+…calling `foo()` returns a promise that *rejects* with `Error('bar')`.
+
+## Example: streaming a response
+
+The benefit of async functions increases in more complex examples. Say you wanted
+to stream a response while logging out the chunks, and return the final size.
+
+{% Aside %}
+The phrase "logging out the chunks" made me sick in my mouth.
+{% endAside %}
+
+Here it is with promises:
+
+```js
+function getResponseSize(url) {
+  return fetch(url).then(response => {
+    const reader = response.body.getReader();
+    let total = 0;
+
+    return reader.read().then(function processResult(result) {
+      if (result.done) return total;
+
+      const value = result.value;
+      total += value.length;
+      console.log('Received chunk', value);
+
+      return reader.read().then(processResult);
+    })
+  });
+}
+```
+
+Check me out, Jake "wielder of promises" Archibald. See how I'm calling
+`processResult()` inside itself to set up an asynchronous loop? Writing that made
+me feel *very smart*. But like most "smart" code, you have to stare at it for
+ages to figure out what it's doing, like one of those magic-eye pictures from
+the 90's.
+
+Let's try that again with async functions:
+
+```js
+async function getResponseSize(url) {
+  const response = await fetch(url);
+  const reader = response.body.getReader();
+  let result = await reader.read();
+  let total = 0;
+
+  while (!result.done) {
+    const value = result.value;
+    total += value.length;
+    console.log('Received chunk', value);
+    // get the next result
+    result = await reader.read();
+  }
+
+  return total;
+}
+```
+
+All the "smart" is gone. The asynchronous loop that made me feel so smug is
+replaced with a trusty, boring, while-loop. Much better. In future, you'll get
+[async iterators](https://github.com/tc39/proposal-async-iteration){: .external},
```

Comment:
```suggestion
[async iterators](https://github.com/tc39/proposal-async-iteration),
```
Please remove all these `{: .external}` things. They don't do anything on web.dev and they are breaking the build.

Reply:
Removed.